src/app/pages/HomePage/ParaTimeSelector/Graph/index.tsx

```diff
@@ -62,25 +71,102 @@ const GraphStyled = styled('svg', {
 }))
 
 const GraphCmp: ForwardRefRenderFunction<SVGSVGElement, GraphProps> = (
-  { disabled = false, transparent = false, selectedGraphEndpoint, setSelectedGraphEndpoint },
+  { disabled = false, transparent = false, selectedGraphEndpoint, setSelectedGraphEndpoint, scale },
   ref,
 ) => {
   const { t } = useTranslation()
   const navigate = useNavigate()
+  const theme = useTheme()
+  const isMobile = useMediaQuery(theme.breakpoints.down('sm'))
+  const enabledGraphEndpoints = useConstant(() => GraphUtils.getEnabledGraphEndpoints())
+  const consensusRef = useRef<SVGCircleElement>(null)
+  const consensusRefInnerCircle = useRef<SVGCircleElement>(null)
+  const emeraldRef = useRef<SVGGElement>(null)
+  const sapphireRef = useRef<SVGGElement>(null)
+  const cipherRef = useRef<SVGGElement>(null)
+  const { setShowMobileTooltip } = useMobileTooltip()
+
+  const [showEmeraldTooltip, setShowEmeraldTooltip] = useState(false)
+
+  const [emeraldRect, emeraldRectRefresh] = useGetBoundingClientRect(emeraldRef)
+  const [sapphireRect, sapphireRectRefresh] = useGetBoundingClientRect(sapphireRef)
+  const [cipherRect, cipherRectRefresh] = useGetBoundingClientRect(cipherRef)
+  const [consensusRect, consensusRectRefresh] = useGetBoundingClientRect(consensusRef)
+  const [consensusInnerCircleRect, consensusInnerCircleRectRefresh] =
+    useGetBoundingClientRect(consensusRefInnerCircle)
+
+  useEffect(() => {
+    emeraldRectRefresh()
+    sapphireRectRefresh()
+    cipherRectRefresh()
+    consensusRectRefresh()
+    consensusInnerCircleRectRefresh()
+  }, [
+    scale,
+    emeraldRectRefresh,
+    sapphireRectRefresh,
+    cipherRectRefresh,
+    consensusRectRefresh,
+    consensusInnerCircleRectRefresh,
+  ])
+
+  useEffect(() => {
+    setShowEmeraldTooltip(false)
```

Comment:
Why we need additional specific state for Emerald only here ? What diff between setShowEmeraldTooltip and setShowMobileTooltip(GraphEndpoints.Emerald, false) ?

Reply:
For "enabled" ParaTimes, the tooltip should trigger upon zooming in. That's why the separate state.